app/controllers/entities_controller.rb

```diff
@@ -0,0 +1,37 @@
+class EntitiesController < ApplicationController
+  before_action :authenticate_user!
+  def index
+    @group = current_user.groups.find(params[:group_id])
+    @entities = @group.entities
+  end
+
+  def new
+    @group = current_user.groups.find(params[:group_id])
+    @entity = @group.entities.new
+  end
+
+  def create
+    @group = current_user.groups.find(params[:group_id])
+    @entity = current_user.entities.create(entity_params)
+    puts @entity
+    if @entity.save
+      @goup_entity = @entity.group_entities.create(group_id: @group.id, entity_id: @entity.id)
+      if @goup_entity.save
+        flash[:notice] = 'New transaction created successfully'
+        redirect_to group_entities_path(@group)
+      else
+        flash.now[:alert] = 'Transaction category creation failed'
+        render action: 'new'
+      end
+    else
+      flash.now[:alert] = 'Transaction creation failed'
+      render action: 'new'
```

Comment:
- [ ] I think these flash messages are not clear enough. Let the message be clear to the user. Your app should be user-friendly. You could ask the user to double-check his fields. With this message: 'Transaction creation failed', I thought there was an issue with your app.